### PR TITLE
pox-3 tests

### PR DESCRIPTION
**Pull request by obycode**
This branch wikll begin to add the tests for the new pox-3 contract.

### CODE REVIEW

File: tests/integration/pox/pox-3/direct-and-pooled.spec.ts [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+    let poxInfo = await getPoxInfo(network);
+    console.log(poxInfo);
+    // Assert that the current cycle has 80m STX locked
+    // and no STX locked for next cycle
+    expect(poxInfo.current_cycle.id).toBe(3);
+    expect(poxInfo.current_cycle.stacked_ustx).toBe(80_000_000_000_000);
+    expect(poxInfo.current_cycle.min_threshold_ustx).toBe(29_290_000_000_000);
+    expect(poxInfo.next_cycle.stacked_ustx).toBe(0);
+    expect(poxInfo.current_cycle.is_pox_active).toBe(true);
```
Comment (obycode) [inline tests/integration/pox/pox-3/direct-and-pooled.spec.ts:239]: This fails. Shouldn't this be true?

File: tests/integration/pox/pox-3/direct-and-pooled.spec.ts [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+    // move on to the next cycle after unlock (N+5)
+    await waitForNextRewardPhase(network, orchestrator, 5);
+
+    let poxInfo = await getPoxInfo(network);
+    // Assert that the current cycle has 900m STX locked from direct stacking
+    expect(poxInfo.current_cycle.id).toBe(2);
+    expect(poxInfo.current_cycle.stacked_ustx).toBe(900_000_000_000_000);
+    // FIXME: Is this right? Alice has stacked 75M and she gets unlocked
+    expect(poxInfo.current_cycle.min_threshold_ustx).toBe(75_000_000_000_000);
```
Comment (obycode) [inline tests/integration/pox/pox-3/direct-and-pooled.spec.ts:157]: This needs to be double-checked. The min threshold is reported as 75M. Alice gets auto-unlocked with 75M.

**Comment by obycode** [comment]
I also still need to update existing tests for changes in the helper functions.

File: tests/integration/pox/helpers.ts
Comment (obycode) [inline tests/integration/pox/helpers.ts:1]: I also still need to update existing tests for changes in the helper functions.

File: tests/integration/pox/pox-3/pooled-stacking-switch-pools.spec.ts
Comment (obycode) [inline tests/integration/pox/pox-3/pooled-stacking-switch-pools.spec.ts:1]: This one does not currently pass.

Comment (obycode) [reply tests/integration/pox/pox-3/pooled-stacking-switch-pools.spec.ts:1]: I am modifying this to test the behavior we are now expecting with pox-3, where a cool down cycle is required between delegating to different pools.

Comment (friedger) [reply tests/integration/pox/pox-3/pooled-stacking-switch-pools.spec.ts:1]: Good to have a test that shows the change in behaviour

File: tests/integration/pox/pox-3/direct-and-pooled.spec.ts (hunk 1 above)
Comment (friedger) [reply tests/integration/pox/pox-3/direct-and-pooled.spec.ts:239]: It used to pass. But the pox activation is not clear to me. Are we in the correct poch?

File: tests/integration/pox/pox-3/pooled-stacking-switch-pools.spec.ts
Comment (obycode) [reply tests/integration/pox/pox-3/pooled-stacking-switch-pools.spec.ts:1]: Updated in d370eff

**Pull request merged**